Adding time-dependent MG to PyBird

billwright93 opened the pull request:
This PR is to implement time-dependent MG (e.g. nDGP gravity) into the existing PyBird code.

billwright93 committed:
Adding time-dep MG expressions for curly G/U/V.

billwright93 committed:
First attempt to add internal numerical computation of growth in nDGP.

billwright93 committed:
First attempt at modifying __init__ for GreenFunction class.

billwright93 committed:
Adding PDF note on how expressions in greenfunction.py need to be modified to include general scale-independent MG.

billwright93 committed:
Added nDGP expressions for mu2(a) and mu22(a).

guidodamico reviewed (changes requested):
This looks good, thank you. I have put a couple of comments in the code, could you please have a look when you have time?

billwright93 committed:
Corrected _MG typo in second- and third-order coefficient functions.

billwright93 committed:
Removed two comments about GR limits for mu2(a) and mu22(a).

guidodamico reviewed (commented):
I answered some questions. Everything looks good, but I'm a bit worried by the precision of the numerical integration.

billwright93 committed:
Made a further attempt at implementing the correct logic for MG and nDGP in pybird.py, bird.py, common.py, and greenfunction.py. Still need to confirm whether the automatic turning on of MG and nDGP when only Omega_rc is specified is working correctly throughout -- may need to move it from greenfunction.py to pybird.py. Still need to add either a proper error or a default Omega_rc value for when nDGP is selected but Omega_rc is not specified. There are many print() statments to remove at a later stage.

billwright93 committed:
Corrected two typos with the call of D_DD_minus_MG_num in greenfunction.py.

billwright93 committed:
Found the correct way to define a function that will be called by scipy.integrate.odeint() while still being a member function of the GreenFunction class. Removed repeated functions that I had previously thought were necessary. Also added missing self. prefixes from two instances of G2d and G2t.

billwright93 committed:
Realised I need to decrease a_ini in the numerical computation of the nDGP growth so that it is always less than the value of a.

billwright93 committed:
Adding iPython notebook to test numerical solution to nDGP growth/decay and the rest of the nDGP implementation.

billwright93 committed:
Merging up-to-date dev branch into WP7_nDGP branch. Resolved conflicts by favouring dev branch or combining modifications.

MarcoMarinucci committed:
Added the nDGP modifications. Solution of differential equation in greenfunction.py, Right normalization for PS in pybird.py.

MarcoMarinucci committed:
Correcting Linear PS

billwright93 committed:
Re-adding Pierre's latest changes from bird.py and pybird.py in the dev branch that had been overwritten in the previous two commits.

billwright93 committed:
Remove debug print statement in bird.py.

billwright93 committed:
Remove debug print statements in greenfunction.py.

MarcoMarinucci committed:
Add files via upload

Last version of nDGP implemented PyBird. This version is currently used for MCMC analysis.

MarcoMarinucci committed:
Create first.gitkeep

MarcoMarinucci committed:
Rename angular.py to nDGP/angular.py

MarcoMarinucci committed:
Rename bird.py to nDGP/bird.py

MarcoMarinucci committed:
Rename first.gitkeep to .gitkeep

MarcoMarinucci committed:
Rename Limber.py to nDGP/Limber.py

MarcoMarinucci committed:
Rename common.py to nDGP/common.py

MarcoMarinucci committed:
Rename fftlog.py to nDGP/fftlog.py

MarcoMarinucci committed:
Rename greenfunction.py to nDGP/greenfunction.py

MarcoMarinucci committed:
Rename likelihood_class.py to nDGP/likelihood_class.py

MarcoMarinucci committed:
Rename nonlinear.py to nDGP/nonlinear.py

MarcoMarinucci committed:
Rename nonlinearLimber.py to nDGP/nonlinearLimber.py

MarcoMarinucci committed:
Rename projection.py to nDGP/projection.py

MarcoMarinucci committed:
Rename pybird.py to nDGP/pybird.py

MarcoMarinucci committed:
Rename resum.py to nDGP/resum.py

MarcoMarinucci committed:
Rename resumfactor.py to nDGP/resumfactor.py

MarcoMarinucci committed:
Create README.md
